dump/pldm_utils.cpp

```diff
@@ -66,28 +100,42 @@ int openPLDM()
         log<level::ERR>(fmt::format("pldm_open failed, errno({}), FD({})", e,
                                     static_cast<int>(fd))
                             .c_str());
-        elog<NotAllowed>(Reason("Required host dump action via pldm is not "
-                                "allowed due to pldm_open failed"));
+        elog<NotAllowed>(Reason("Failed to open PLDM"));
     }
     return fd;
 }
 
-uint8_t getPLDMInstanceID(uint8_t eid)
+pldm_instance_id_t getPLDMInstanceID(uint8_t tid)
 {
-    constexpr auto pldmRequester = "xyz.openbmc_project.PLDM.Requester";
-    constexpr auto pldm = "/xyz/openbmc_project/pldm";
-
-    auto bus = sdbusplus::bus::new_default();
-    auto service = internal::getService(bus, pldm, pldmRequester);
+    pldm_instance_id_t instanceID = 0;
 
-    auto method = bus.new_method_call(service.c_str(), pldm, pldmRequester,
-                                      "GetInstanceId");
-    method.append(eid);
-    auto reply = bus.call(method);
+    auto rc = pldm_instance_id_alloc(pldmInstanceIdDb, tid, &instanceID);
+    if (rc == -EAGAIN)
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Would it help to log a message here? just to know that this is cause of 100 millisecond delay